ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,256 @@
+- **namespace** - The rego namespace of the test policy. 
+- **successes** - The number of successful checks in the form of an integer
+- **failures** - A JSON list containing objects describing each failure
+
+The **failures** list of objects will provide the following information about each check:
+- **name** - Name of the individual check written in [snake case](https://en.wikipedia.org/wiki/Snake_case)
+- **msg** - Message about what went wrong with the check
+- **description** - The explanation about why the check is significant (Why the label is used, why the vulnerabilities are checked etc.)
+- **url** - link to the further documentation on the individual check (or that type of check)
+
+Example full JSON test output for a Tekton task that tests the container image labels:
+```
+[
+    {
+        "filename": "image-inspect.json",
+        "namespace": "image_labels",
+        "successes": 19,
+        "failures": [
+            {
+                "msg": "The 'architecture' label is required",
+                "metadata": {
+                    "details": {
+                        "description": "Architecture the software in the image should target.",
+                        "name": "architecture_label_required",
+                        "url": "https://source.redhat.com/groups/public/container-build-system/container_build_system_wiki/guide_to_layered_image_build_service_osbs#jive_content_id_Labels"
+                    }
+                }
+            }
+          ]
+    }
+]
+```
+
+### Information injection
+
+Whenever possible, resources like ConfigMaps or Secrets will be used to inject configuration into Tasks. This is preferred to templates and patches as it fits well with Kubernetes declarative and GitOps approaches.
+tf
+ConfigMaps and Secrets will be mounted into the Task pod to inject file-based information like certificates.
+Environment variables may be injected from ConfigMaps.
+
+Variables directly configured in the Pod definition are discouraged.
+
+Side note: Secrets and ConfigMaps should be made [immutable](https://kubernetes.io/docs/concepts/configuration/secret/#secret-immutable)
+
+
+### Information format
+
+[Clearly define the format of input parameters and results](https://github.com/tektoncd/catalog/blob/main/recommendations.md#clearly-define-the-format-of-input-parameters-and-results).
+
+Atomic information may be passed as a simple parameter. Non binary encoded complex information should be exchanged through [JSON format](https://github.com/tektoncd/catalog/blob/main/recommendations.md#use-composable-parameter-formats) between Tasks.
+
+#### Image references
+
+Tags are not uniquely identifying an image so that we cannot rely on them. They do not guarantee that the scanning is for the image built as part of the PipelineRun if there is a race with a parallel build. As such, image digests will be used for image references.
```

Comment:
Digests can be rendered invalid if there is no tag pointing to them, so if you are concerned about the tag moving to a different digest, then the digest references will still end up being invalid.

I am not disagreeing with the use of digest, just disagreeing on the logic to defend the choice.

Reply:
Thanks! I update it as below, is it ok?
```
Tags are not uniquely identifying an image so that we cannot rely on them. They do not guarantee that the scanning is for the image built as part of the PipelineRun if there is a race with a parallel build. As such, image digests will be used for image references then we can get an error once the original image doesn't exist any more.
```

Reply:
How about something like

> Since tags can be moved from one image to another, they should not be relied on as a reference. In order to guarantee that any scanning is performed on an image built as part of a PipelineRun, the immutable image digest reference will be used instead.

The fact that digests can become invalid if all tags are removed is an implementation detail of the registry. Hopefully that won't happen to people, but I don't think it has any bearing on the test workstream if it does.

Reply:
Thanks!